src/pybind/mgr/orchestrator.py

```diff
@@ -333,6 +333,18 @@ def __init__(self):
         # The type of service (osd, mon, mgr, etc.)
         self.service_type = None
 
+        # Service version that was deployed
+        self.version = None
+
+        # Location of the configuration of the service, it might be
+        # a filesystem path, or a rados object path (pool/name).
```

Comment:
If it's a _filesystem path_, this is going to be a host dependent path with no meaning on other systems. I'd be :-1: for exposing filesystem paths of remote machines here. This should IMO be an implementation detail of the external orchestrator and not published to the cli.

Reply:
Fair enough. Then what if I rename the attribute to `rados_config_location` and it's just meant for rados objects?

Reply:
Sounds good!

Reply:
This is a specific instance of a service, running on a particular host, so isn't it OK for a filesystem path to be host-dependent?

Reply:
Sure it is.

But why do we want to expose them? No one, except for the external orchestrator, should directly touch that file. No user should ssh into a running container and modify that file.

Reply:
> No user should ssh into a running container and modify that file.

The orchestrator is not meant just for container environments, right?
Having the location of where the orchestrator implementation stored the configuration of a particular service instance might be useful for sysadmins to debug problems.

FYI, in my current use case, I only store rados objects paths in this attribute.

Reply:
Fair enough. That raises the question though, if external orchestrators are the only things meant to touch service configuration, why expose configuration that's stored in rados objects to orchestrator clients? Doesn't that blur the lines between what's in control of service configuration (external orchestrator vs. orchestrator client)?

Reply:
For rados objects paths, we have a real use case: the dashboard can directly set the reados objects without going through the external ouchestrator.

Reply:
What I understood from all the discussions about what are the responsibilities of the orchestrator was that the orchestrator is only responsible for deployment/undeployment. The management of running services was done by external tools, such as the dashboard.

That is why I added this attributes to the `ServiceDescription` class, to allow external tools to take over after the orchestrator deploys the services.

Reply:
> What I understood from all the discussions about what are the responsibilities of the orchestrator was that the orchestrator is only responsible for deployment/undeployment.

Depends on the orchestrator and the service, e.g. DeepSea knows how to create RGW users, which is arguably management of a running service, but that you would of course also want to do through the dashboard.

Anyway, it's all good, I understand the rationale now, so fine by me :-)